src/Microsoft.ML.AutoML/TrainerExtensions/SweepableParams.cs

```diff
@@ -115,6 +117,15 @@ public static IEnumerable<SweepableParam> BuildLightGbmParams()
             };
         }
 
+        public static IEnumerable<SweepableParam> BuildMatrixFactorizationParmas()
+        {
+            return new SweepableParam[]
+            {
+                new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.NumberOfIterations), new object[] { 10, 20 }),
+                new SweepableFloatParam(nameof(MatrixFactorizationTrainer.Options.LearningRate), 0.025f, 0.04f, numSteps:2),
+                new SweepableDiscreteParam(nameof(MatrixFactorizationTrainer.Options.ApproximationRank), new object[] { 10, 20 }),
```

Comment:
Please copy sweep ranges & hyperparameters from:
https://github.com/dotnet/machinelearning/blob/edfd10f131abe97a7f7bb25c0070843a67b8b660/src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs#L173-L250

Ideally, in the future, we should access them directly from the learner instead of having our copy in AutoML.

Reply:
Seems that automl will try 5*4*3*3*... set of params on MatrixFactorization, will that cost too much time?

Reply:
tldr; SMAC focuses our search efforts, and we are bounded by our total runtime.

----

This is expected (and should be greater). The Bayesian hyperparameter optimization (SMAC) focuses on the useful areas of the search space. 

For selected trainers, we first do 20 iterations of random sweeping to warm up the search space. Then SMAC uses the found results of those 20 iterations to predict which areas of the hyperparameter space is best to explore next. The choice of what to explore next is based on which areas areas are doing best and which areas are unexplored. 

Beyond SMAC we have additional hyperparameter optimization algos we should be using like [KDO](https://github.com/dotnet/machinelearning/blob/master/src/Microsoft.ML.Sweeper/Algorithms/KdoSweeper.cs).

Reply:
For your comment in the PR description:
>  figuring out how to do sweepParams for MatrixFactorization, the current one seems to have some bugs and it's never stop! (maybe SMACSweeper does not converge?)

Is AutoML not stopping at the specified timeout? If you set it to 60s of runtime, it should stop soon after this limit.

AutoML is designed to do round robin between three trainers culled from 8-11 depending on the task. Since this has only one trainer to choose from, perhaps the AutoML code needs updating?

Reply:
Turns out that I forget to set the experiment time so it is defaulted to 16400... My Bad